Assets/Scripts/Conductor.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Conductor : MonoBehaviour
+{
+    public AudioSource Song;
+
+    private float _bpm;
+    private float _crotchet;
+    private float offset;
+    private double _dspTimeSong;
+    private float _lastBeat;
+
+    private List<BeatListener> _listeners = new();
+
+    public void Start()
+    {
+        _bpm = 156;
+        _crotchet = 60 / _bpm;
+        offset = 0;
+        _dspTimeSong = 0;
+        _lastBeat = 0;
+
+        Song = GetComponent<AudioSource>();
+        Song.Play();
+        OnSongStart();
+    }
+
+    public void Update()
+    {
+        if(OnHalfBeat(_lastBeat))
+        {
+            foreach(var listener in _listeners)
+            {
+                listener.OnHalfBeat();
+            }
+
+            _lastBeat = IncrementLastBeat(_lastBeat);
+        }
+    }
+
+    public void RegisterListener(BeatListener listener)
+    {
+        _listeners.Add(listener);
+    }
+
+    public void OnSongStart()
+    {
+        _dspTimeSong = AudioSettings.dspTime;
+    }
+
+    public float BeatsToSeconds(float beats)
+    {
+        return beats * _crotchet;
+    }
+
+    public float GetSongPosition()
+    {
+        return (float)(AudioSettings.dspTime - _dspTimeSong) * Song.pitch - offset;
+    }
+
+    public bool OnHalfBeat(float check)
+    {
+        return check + _crotchet / 2 < GetSongPosition();
+    }
+
+    public float IncrementLastBeat(float time)
+    {
+        return time + _crotchet / 2;
+    }
+
+    public float HalfBeatsToSeconds(int numBeats)
+    {
+        return numBeats * _crotchet;
+    }
+}
```

Comment:
We need to cleanup naming. 
This calculates beats to seconds, not half beats to seconds. 
 Maybe we should use floats so we can get partial beats? 
 - Quarter beats would be 0.25, half would be 0.5, etc..